### DISCUSSION TITLE
How to point JCL to dcc32

### DISCUSSION THREAD

**Discussion by notalfredo** [project-jedi/jcl#150] [Q&A] [answered]
Hello I am trying to do non-manual installation of this library.

When I ran this installation verbose I got this message

```
Initializing/Updating git submodules...

===================================================================
Compiling JediIncCheck...
Delphi 7 is no valid installation
 - dcc32.exe missing (Evaluation version and TurboExplorer are not supported)
 - System.dcu missing

Using Embarcadero RAD Studio 10.6
Using search path: C:\Program Files (x86)\Embarcadero\Studio\23.0\lib\win32\release

This version of the product does not support command line compiling.


An error occurred while compiling the installer. Installation aborted.

Press any key to continue . . .
```

A couple notes. 

A long long time ago I had a valid version of Delphi 2007 but I got rid of it some time ago. But recently I needed a valid version of Delphi 2010. I first installed a free trail of Delphi CE then shortly after I purchased a valid license so I can get access to Delphi 2010.

[By looking at other questions asked here](https://github.com/project-jedi/jcl/discussions/145) it says that `Neither the community version, nor the trial version provide the command line compiler (dcc32) that is used by the JCL installer.`

But the thing is that I paid for the professional version. 

I know I have access to `dcc32` because if I run `dcc32` in my terminal I get this output 

```
PS C:\Users\Alfredo\Desktop> dcc32
Embarcadero Delphi for Win32 compiler version 21.0
Copyright (c) 1983,2009 Embarcadero Technologies, Inc.

Syntax: dcc32 [options] filename [options]

  -A<unit>=<alias> = Set unit alias
  -B = Build all units
  -CC = Console target
  -CG = GUI target
  -D<syms> = Define conditionals
  -E<path> = EXE/DLL output directory
  -F<offset> = Find error
  -GD = Detailed map file
  -GP = Map file with publics
  -GS = Map file with segments
  -H = Output hint messages
  -I<paths> = Include directories
  -J = Generate .obj file
  -JPHNE = Generate C++ .obj file, .hpp file, in namespace, export all
  -JL = Generate package .lib, .bpi, and all .hpp files for C++
  -K<addr> = Set image base addr
  -LE<path> = package .bpl output directory
  -LN<path> = package .dcp output directory
  -LU<package> = Use package
  -M = Make modified units
  -N0<path> = unit .dcu output directory
  -NH<path> = unit .hpp output directory
  -NO<path> = unit .obj output directory
  -NB<path> = unit .bpi output directory
  -NS<namespaces> = Namespace search path
  -O<paths> = Object directories
  -P = look for 8.3 file names also
  -Q = Quiet compile
  -R<paths> = Resource directories
  -U<paths> = Unit directories
  -V = Debug information in EXE
  -VR = Generate remote debug (RSM)
  -W[+|-|^][warn_id] = Output warning messages
  -Z = Output 'never build' DCPs
  -$<dir> = Compiler directive
  --help = Show this help screen
  --version = Show name and version
  --codepage:<cp> = specify source file encoding
  --default-namespace:<namespace> = set namespace
  --depends = output unit dependency information
  --doc = output XML documentation
  --drc = output resource string .drc file
  --no-config = do not load default DCC32.CFG file
  --description:<string> = set executable description
  --inline:{on|off|auto} = function inlining control
  --string-checks:{on|off} = string format checking (default: on)
  --peflags:<flags> = set extra PE Header flags field
  --peoptflags:<flags> = set extra PE Header optional flags field
  --peosversion:<major>.<minor> = set OS Version fields in PE Header (default: 5.0)
  --pesubsysversion:<major>.<minor> = set Subsystem Version fields in PE Header (default: 4.0)
  --peuserversion:<major>.<minor> = set User Version fields in PE Header (default: 0.0)
Compiler switches: -$<letter><state> (defaults are shown below)
  A8  Aligned record fields
  B-  Full boolean Evaluation
  C+  Evaluate assertions at runtime
  D+  Debug information
  G+  Use imported data references
  H+  Use long strings by default
  I+  I/O checking
  J-  Writeable structured consts
  L+  Local debug symbols
  M-  Runtime type info
  O+  Optimization
  P+  Open string params
  Q-  Integer overflow checking
  R-  Range checking
  T-  Typed @ operator
  U-  Pentium(tm)-safe divide
  V+  Strict var-strings
  W-  Generate stack frames
  X+  Extended syntax
  Y+  Symbol reference info
  Z1  Minimum size of enum types
  ```
  
Now if by this terminal output it means that I do indeed have the `dcc32` compiler I would like to know how I can switch it over to use that one.
  
 
While I wait I will see if I can figure that out. Worst case I will do a manual installation

**Comment by obones** [comment]
AFAIR, it's all read from the registry.
It appears as if the installer sees Delphi 7, and an installation of Rad Studio 12 but not the D2010 version that you seem to think of having.
And yet, when you run dcc32, it shows a version that is Delphi XE7 which is not D2010

There is something very strange on your computer, a mixup of various versions that makes it really hard to figure out what's wrong and how to fix it.

**Reply by notalfredo** [reply]
Okay thanks for the quick reply. Im confident I have Delphi 2010 installed because I can open it and make programs in it. BUT I will see what I can figure out. If I found an answer ill post it here. Worst case I do a manual installation. 

Is there a way do something like `dcc32 <bat-file>` ? I tried doing that but it exploded it on me. Anyways ...

Thank you!

**Reply by ronaldhoek** [reply, marked as answer]
Not sure, but you could try just compiling the 'JediInstaller.dpr' project because that's the main part of the non-manual installation process (but i'bve nver done that... so I'm not sure if it works at all))

**Comment by notalfredo** [comment]
> Not sure, but you could try just compiling the 'JediInstaller.dpr' project because that's the main part of the non-manual installation process (but i'bve nver done that... so I'm not sure if it works at all))

I got it working because of this. 
Cheers !
